File: MDM-Web.API.Model/Device.cs
#nullable enable
using System;

namespace Model
{
    public class Device
    {
        public Guid Id { get; set; }
        public string deviceID { get; set; }
        public string? deviceToken { get; set; }
        public string? Owner { get; set; }
        public int BatteryState { get; set; }
        
        public float BatteryLevel { get; set; }
        
        public string Name { get; set; }
        
        public string SystemName { get; set; }
        
        public string SystemVersion { get; set; }
        
        public string Model { get; set; }
        
    }
}

File: MDM-Web.React/views/src/Components/Device.js
import React, {useState, useEffect} from "react";
import {useParams, useHistory} from "react-router-dom";

export default function Device() {
    const {id} = useParams();
    const history = useHistory()
    const [device, setDevice] = useState(null);

    useEffect(() => {
        function loadDevice() {
            return {
                deviceID: 1,
                Name: "Urządzenie Filipa",
                SystemName: "IOS",
                SystemVersion: "13",
                Model: "Iphone 11",
                BatteryState: 99,
                BatteryLevel: 69
            }
        }

        function onLoad() {
            try {
                const device = loadDevice();
                setDevice(device);
            } catch (e) {
                alert("Nie udało się wczytać urządzenia")
            }
        }

        onLoad();
    }, [id])

    return (
        <div className="Device">
            {device.Name}
        </div>
    )
}